### PR TITLE
Branchlist pullrequest sha clipboard

### PR DESCRIPTION
**Pull request by jessieyu777**
Migrate a commit-copy-action as a component used in the branchlist page & pull-request page
Screenshot 2024-10-04 at 2 36 18 PM
Screenshot 2024-10-04 at 2 36 38 PM
Screenshot 2024-10-04 at 2 36 47 PM

### CODE REVIEW

**Review by jessieyu777 — COMMENTED** [review]
made copied board as a component used in pr-commit-page & branch-list page

File: packages/playground/src/components/commit-copy-actions.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      clearTimeout(timeoutId)
+    }
+  }, [copied])
+
+  return (
+    <ShaBadge.Root>
+      <ShaBadge.Content>
+        <Link to="#">
+          <Text size={1} className="text-tertiary-background">
+            {sha.substring(0, 6)}
```
Comment (vardanbansal-harness) [inline packages/playground/src/components/commit-copy-actions.tsx:27]: Do `sha.slice(0, 7)` instead

File: packages/playground/src/components/pull-request/pull-request-commits.tsx [hunk 2]
```diff
@@ -1,49 +1,13 @@
...
 import { TypesCommit } from './interfaces'
 import { formatDate, getInitials } from '../../utils/utils'
-import { StackedList, Text, NodeGroup, Avatar, AvatarFallback, Icon, cn } from '@harnessio/canary'
-import copy from 'clipboard-copy'
-import { ShaBadge } from '../..'
-import { Link } from 'react-router-dom'
-
+import { StackedList, Text, NodeGroup, Avatar, AvatarFallback } from '@harnessio/canary'
+import { CommitCopyActions } from '../commit-copy-actions'
+//Icon, cn
```
Comment (cjlee01) [inline packages/playground/src/components/pull-request/pull-request-commits.tsx:6]: is this comment needed?

**Review by cjlee01 — APPROVED** [review]

File: packages/playground/src/components/pull-request/pull-request-commits.tsx (hunk 2 above)
Comment (jessieyu777) [reply packages/playground/src/components/pull-request/pull-request-commits.tsx:6]: Removeed

**Review by jessieyu777 — COMMENTED** [review]
revise it based on comments

**Pull request merged**